tv/src/main/java/butter/droid/tv/ui/trailer/TVTrailerPlayerActivity.java

```diff
@@ -111,97 +107,25 @@ public void onDisableVideoPlayerSubsButton() {
     }
 
     @Override
-    public void onExecuteQueryYoutubeTask(String youtubeUrl) {
-        final QueryYouTubeTask youTubeTask = new QueryYouTubeTask(this, youTubeManager, networkManager);
-        final String youTubeVideoId = youTubeManager.getYouTubeVideoId(youtubeUrl);
-        youTubeTask.execute(youTubeVideoId);
-    }
-
-    @Override
     public void onNotifyMediaReady() {
         playerFragment.onMediaReady();
     }
 
     @Override
     public void onDisplayErrorVideoDialog() {
-        try {
-            final AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(TVTrailerPlayerActivity.this);
-            alertDialogBuilder.setTitle(R.string.comm_error);
-            alertDialogBuilder.setCancelable(false);
-            alertDialogBuilder.setMessage(R.string.comm_message);
-
-            alertDialogBuilder.setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
-                @Override
-                public void onClick(DialogInterface dialog, int which) {
-                    TVTrailerPlayerActivity.this.finish();
-                }
-            });
-
-            final AlertDialog dialog = alertDialogBuilder.create();
-            dialog.show();
-        } catch (Exception e) {
-            Timber.e("Problem showing error dialog", e);
-        }
+        DialogFactory.createErrorFetchingYoutubeVideoDialog(this, new ActionCallback() {
```

Comment:
In this case, keep and eye that TVTrailerPlayerActivity and TrailerPlayerActivity has both different parent activities (that doesn't share the same inheritance). So for that reason I leaved it here.